File: SubscriptionApi/SubscriptionApi/Queries/OrganizationalUnitQuery.cs
using System.Collections.Generic;
using System.Linq;
using SubscriptionApi.Authorization;

namespace SubscriptionApi.Queries
{
  public class OrganizationalUnitQuery : AssetQuery
  {
    private readonly string _name;
    private readonly AssetQueryResolution _authorizationStructure;

    public OrganizationalUnitQuery(string name, AssetQueryResolution authorizationStructure)
    {
      _name = name;
      _authorizationStructure = authorizationStructure;
    }

    public void ResolveInto(List<string> requestedAssetNames, QueryResolutionEvents resolutionEvents)
    {
      var assetsFromQuery = _authorizationStructure
        .RetrieveAssetsByOrganizationalUnitName(_name);
      if (!assetsFromQuery.Any())
      {
        resolutionEvents.NoResolutionResultsFor(_name);
      }
      else
      {
        requestedAssetNames.AddRange(assetsFromQuery);
      }

    }
  }
}

File: Java/factories/_01_encapsulation_of_global_context/src/main/java/queries/OrganizationalUnitQuery.java
package queries;

import authorization.AssetQueryResolution;

import java.util.List;

public class OrganizationalUnitQuery implements AssetQuery {
    private final String name;
    private final AssetQueryResolution authorizationStructure;

    public OrganizationalUnitQuery(String name, AssetQueryResolution authorizationStructure) {
        this.name = name;
        this.authorizationStructure = authorizationStructure;
    }

    public void resolveInto(List<String> requestedAssetNames, QueryResolutionEvents resolutionEvents) {
        List<String> assetsFromQuery = authorizationStructure
            .retrieveAssetsByOrganizationalUnitName(name);
        if (assetsFromQuery.isEmpty()) {
            resolutionEvents.noResolutionResultsFor(name);
        } else {
            requestedAssetNames.addAll(assetsFromQuery);
        }

    }
}
